src/Engine/ProtoCore/AssociativeGraph.cs

```diff
@@ -16,6 +16,69 @@ public enum UpdateStatus
     public class Utils
     {
         /// <summary>
+        /// Gets the dirty graphnode of the given pc
+        /// </summary>
+        /// <param name="pc"></param>
+        /// <param name="classIndex"></param>
+        /// <param name="procIndex"></param>
+        /// <returns></returns>
+        public static AssociativeGraph.GraphNode GetGraphNodeAtPC(int pc, List<AssociativeGraph.GraphNode> graphNodesInScope)
+        {
+            if (graphNodesInScope != null && graphNodesInScope.Count > 0)
```

Comment:
I'd suggest nullity test could be done by assertion. And `graphNodesInScope.Count > 0` is redundant.